app/main.py

```
@@ -1,52 +1,38 @@
+from app.book import Book
+from app.display import DisplayConsole, DisplayReverse
+from app.printer import ConsolePrinter, ConsoleReversePrinter
+from app.serializer import JSONSerializer, XMLSerializer
 
+_DISPLAYS = {
+    "console": DisplayConsole(),
+    "reverse": DisplayReverse(),
+}
 
-class Book:
-    def __init__(self, title: str, content: str):
-        self.title = title
-        self.content = content
+_PRINTERS = {
+    "console": ConsolePrinter(),
+    "reverse": ConsoleReversePrinter(),
+}
 
-    def display(self, display_type: str) -> None:
-        if display_type == "console":
-            print(self.content)
-        elif display_type == "reverse":
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown display type: {display_type}")
-
-    def print_book(self, print_type: str) -> None:
-        if print_type == "console":
-            print(f"Printing the book: {self.title}...")
-            print(self.content)
-        elif print_type == "reverse":
-            print(f"Printing the book in reverse: {self.title}...")
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown print type: {print_type}")
-
-    def serialize(self, serialize_type: str) -> str:
-        if serialize_type == "json":
-            return json.dumps({"title": self.title, "content": self.content})
-        elif serialize_type == "xml":
-            root = ET.Element("book")
-            title = ET.SubElement(root, "title")
-            title.text = self.title
-            content = ET.SubElement(root, "content")
-            content.text = self.content
-            return ET.tostring(root, encoding="unicode")
-        else:
-            raise ValueError(f"Unknown serialize type: {serialize_type}")
+_SERIALIZERS = {
+    "json": JSONSerializer(),
+    "xml": XMLSerializer(),
+}
 
 
 def main(book: Book, commands: list[tuple[str, str]]) -> None | str:
     for cmd, method_type in commands:
```

If `cmd` is not one of `display`, `print`, or `serialize` the loop body will do nothing for that command (no exception is raised). The `KeyError` except only catches missing method types for recognized commands. The error message raised on KeyError suggests unknown *command or method type*, but currently unknown commands are silently ignored. Consider explicitly handling unknown commands (e.g., raise `ValueError`) to match the error message and avoid silent no-ops, unless tests intentionally rely on the current silent behavior.